configureWorkspace/configure.ts

```diff
@@ -626,29 +651,56 @@ export async function configure(actionContext: IActionContext, rootFolderPath?:
         rootFolderPath = folder.uri.fsPath;
     }
 
-    const platformType: Platform = await quickPickPlatform();
+    return configureCore(
+        actionContext,
+        {
+            rootPath: rootFolderPath,
+            outputFolder: rootFolderPath
+        });
+}
+
+export async function configureApi(actionContext: IActionContext, options: ConfigureApiOptions): Promise<void> {
+    return configureCore(actionContext, options);
+}
+
+// tslint:disable-next-line:max-func-body-length // Because of nested functions
+async function configureCore(actionContext: IActionContext, options: ConfigureApiOptions): Promise<void> {
+    let properties: TelemetryProperties & ConfigureTelemetryProperties = actionContext.properties;
+    let rootFolderPath: string = options.rootPath;
+    let outputFolder = options.outputFolder;
+
+    const platformType: Platform = options.platform || await quickPickPlatform();
     properties.configurePlatform = platformType;
 
-    let os: OS | undefined;
-    if (platformType.toLowerCase().includes('.net')) {
+    let os: OS | undefined = options.os;
+    if (!os && platformType.toLowerCase().includes('.net')) {
         os = await quickPickOS();
     }
     properties.configureOs = os;
 
-    let port: string;
-    if (platformType.toLowerCase().includes('.net')) {
-        port = await promptForPort(80);
-    } else {
-        port = await promptForPort(3000);
+    let port: string = options.port;
+    if (!port) {
+        if (platformType.toLowerCase().includes('.net')) {
+            port = await promptForPort(80);
+        } else {
+            port = await promptForPort(3000);
+        }
     }
 
-    let serviceNameAndRelativePath: string;
-    if (platformType.toLowerCase().includes('.net')) {
-        serviceNameAndRelativePath = await findCSProjFile(rootFolderPath);
-        properties.packageFileType = '.csproj';
-        properties.packageFileSubfolderDepth = getSubfolderDepth(serviceNameAndRelativePath);
-    } else {
-        serviceNameAndRelativePath = path.basename(rootFolderPath).toLowerCase();
+    let serviceNameAndPathRelativeToOutput: string;
+    {
```

Comment:
nit: get rid of unnecessary curly brackets

Reply:
They're there to force serviceNameAndPathRelativeToRoot to go out of scope and not get used after calculating serviceNameAndPathRelativeToRoot. I'll add a comment.